pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -816,6 +822,146 @@ func GatherMachineSet(i *Gatherer) func() ([]record.Record, []error) {
 	}
 }
 
+// GatherInstallPlans collects Top x InstallPlans.
+// Because InstallPlans have unique generated names, it groups them by namespace and the "template"
+// for name generation from field generateName.
+// It also collects Total number of all installplans and all non-unique installplans.
+//
+// The Kubernetes api https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/certificates/v1beta1/certificatesigningrequest.go#L78
+// Response see https://docs.openshift.com/container-platform/4.3/rest_api/index.html#certificatesigningrequestlist-v1beta1certificates
+//
+// Location in archive: config/certificatesigningrequests/
+func GatherInstallPlans(i *Gatherer) func() ([]record.Record, []error) {
+	return func() ([]record.Record, []error) {
+		var plansBatchLimit int64 = 500
+		cont := ""
+		recs := map[string]*collectedPlan{}
+		total := 0
+		opResource := schema.GroupVersionResource{Group: "operators.coreos.com", Version: "v1alpha1", Resource: "installplans"}
+		resInterface := i.dynamicClient.Resource(opResource).Namespace("openshift-operators")
```

Comment:
I just found that there are more installplans than just in `openshift-operators` namespace. I am wondering if this limitation makes sense. I did `oc get installplan -A` and there are more installplans in `openshift-storage`, `openshift-logging`, `default`, etc. (in shared4.5 cluster). WDYT?

Reply:
It is a good idea, I will try to get all.